rfcs/rfcOVID-0019.md

```diff
@@ -0,0 +1,342 @@
+# Built-in lexical exporting
+
+## Preamble
+
+    Author:  Curtis Poe <curtis.poe@gmail.com>
+    Sponsor:
+    ID:      OVID-0019
+    Status:  Draft
+
+## Abstract
+
+This RFC proposes adding built-in lexical exporting via an `:export`
+attribute.
+
+## Motivation
+
+To eliminate the need for the myriad different CPAN modules for exporting
+functions. This RFC is only for functions, not variables.
+
+## Rationale
+
+There are many exporter modules available. The core `Exporter` is the best
+known, and has the curious feature of allowing procedural modules to use
+inheritance to implement exporting via package variables. It has a clumsy
+interface and many alternatives arose:
+
+* [Sub::Exporter](https://metacpan.org/pod/Sub::Exporter)
+* [Exporter::Lite](https://metacpan.org/pod/Exporter::Lite)
+* [Exporter::Easy](https://metacpan.org/pod/Exporter::Easy)
+* [Exporter::Tiny](https://metacpan.org/pod/Exporter::Tiny)
+
+And more, including many modules which manually implement exporting directly
+in the `import` method.
+
+There are also many attribute-based exporters:
+
+* [Exporter::Simple](https://metacpan.org/pod/Exporter::Simple)
+* [Perl6::Export](https://metacpan.org/pod/Perl6::Export) (not really attributes)
+* [Perl6::Export::Attrs](https://metacpan.org/pod/Perl6::Export::Attrs)
+* [Exporter::Attributes](https://metacpan.org/pod/Exporter::Attributes)
```

Comment:
This is my favorite, supports both variables and functions, and also optional exporting albeit with capital `:Export` and `:Exportable` attributes.